### PR TITLE
Taskcluster rework

### PR DESCRIPTION
**Pull request by garbas**

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,34 @@
+FROM alpine
```
Comment (La0) [inline Dockerfile]: Just curious: why alpine here ?

File: lib/please_cli/please_cli/build.py [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+    if app in ['releng-frontend', 'releng-docs', 'shipit-frontend']:
+        code, output =  please_cli.utils.run_command(
+            '{nix_build} {ROOT_DIR}/nix/default.nix -A apps.{app} -o {ROOT_DIR}/result-{app}'.format(  # noqa
+                app=app,
+                nix_build=nix_build,
+                ROOT_DIR=please_cli.config.ROOT_DIR,
+                ))
+    else:
+        code, output =  please_cli.utils.run_command(
+            '{nix_build} {ROOT_DIR}/nix/default.nix -A apps.{app}.docker -o {ROOT_DIR}/result-{app}'.format(  # noqa
```
Comment (La0) [inline lib/please_cli/please_cli/build.py]: Frontend are not build into the docker env ?

File: lib/please_cli/please_cli/config.py [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+
+DEPLOYMENT_BRANCHES = ['staging', 'production', 'taskcluster-rework']
+
+
+NIX_BIN_DIR = ""  # must end with /
+
+with open(os.path.join(os.path.dirname(__file__), 'VERSION')) as f:
+    VERSION = f.read().strip()
+
+# XXX: below data should be placed in src/<app>/default.nix
```
Comment (La0) [inline lib/please_cli/please_cli/config.py]: :100:

File: lib/please_cli/please_cli/maintanance.py [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+from __future__ import absolute_import
+
+import click
+
+
+@click.command()
+@click.argument('apps', nargs=-1)
+def cmd_on(apps):
+    click.echo('TODO: maintanance ON')
```
Comment (La0) [inline lib/please_cli/please_cli/maintanance.py]: Nit: that's maintenance.

File: Dockerfile (hunk 1 above)
Comment (garbas) [inline Dockerfile]: I was building on top of `nixos/nix` image ([Dockerfile](https://github.com/NixOS/nix/blob/master/misc/docker/Dockerfile)) and extending it with our stuff.

In final version (later today) there wont be any Dockerfile at the root of the repo, but there will be a command to rebuild (and push) base image.

File: lib/please_cli/please_cli/build.py (hunk 2 above)
Comment (garbas) [inline lib/please_cli/please_cli/build.py]: yeah, i hacked something to get things passing. now i need to solve this for every scenario. also hooks are missing, which i'm adding shortly

**Pull request merged**